kornia/augmentation/base.py

```diff
@@ -190,8 +193,7 @@ def forward(self, input: Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]],
                 params: Optional[Dict[str, torch.Tensor]] = None,  # type: ignore
                 return_transform: Optional[bool] = None
                 ) -> Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]]:  # type: ignore
-        in_tensor, in_transform = self.__infer_input__(input)
-        batch_shape = in_tensor.shape
+        in_tensor, batch_shape, in_transform = self.__infer_input__(input)
```

Comment:
Why not just use the ```batch_shape```? We do not need to alter the functional part.

```python
in_tensor, batch_shape, in_transform = self.__infer_input__(input)
...SOME_LOGIC...
out_shape = out_tensor.shape

if len(out_shape) == 4:  #"2D"
    batch_shape[-2:] = out_shape[-2:]

if len(out_shape) == 5:  #"3D"
    batch_shape[-3:] = out_shape[-3:]

out_tensor.view(*batch_shape)
```

Reply:
Also, I think the transformation matrix needs to do the same as well. In case user inputs (3, 3) than (1, 3, 3)

Reply:
One more thing that I suggest not to perform the transformation inside the ```__infer_input__```.

Do it outside, then we may not need to have another ```_transform_shape```. It could be confusing for having similar names.

Reply:
> Why not just use the `batch_shape`? We do not need to alter the functional part.
> 
> ```python
> in_tensor, batch_shape, in_transform = self.__infer_input__(input)
> ...SOME_LOGIC...
> out_shape = out_tensor.shape
> 
> if len(out_shape) == 4:  #"2D"
>     batch_shape[-2:] = out_shape[-2:]
> 
> if len(out_shape) == 5:  #"3D"
>     batch_shape[-3:] = out_shape[-3:]
> 
> out_tensor.view(*batch_shape)
> ```

I need some clarification. Are we not going to modify the functional part, only the `_AugmentationBase`? 
All the functions in functional modify the shape too.

> Also, I think the transformation matrix needs to do the same as well. In case user inputs (3, 3) than (1, 3, 3)

sorry where is this transformation matrix?



> One more thing that I suggest not to perform the transformation inside the `__infer_input__`.
> 
> Do it outside, then we may not need to have another `_transform_shape`. It could be confusing for having similar names.

I agree on this.

Reply:
> Another proposal, how about having a `keep_dims` param to give the user a choice?

Sorry I missed this comment. I think this is a good way. Just to confirm, I will keep the decorator and add `keepdim` to each function in functional right?

Reply:
>  I need some clarification. Are we not going to modify the functional part, only the `_AugmentationBase`? 

Yes. Low-level API will only input BCHW and output BCHW. The shape transformation shall be in the high level modules. 

> All the functions in functional modify the shape too.

You are right. It should not. Please replace the shape transformation into a shape check.

> sorry where is this transformation matrix?

When the input/output is a (tensor, tensor) tuple, the second tensor shall be the corresponding transformation matrix.


> Sorry I missed this comment. I think this is a good way. Just to confirm, I will keep the decorator and add keepdim to each function in functional right?

Not functional changes. We will only add the high-level shape control inside the `_AugmentationBase`.

Reply:
ok got it!